src/ln/channelmonitor.rs

```diff
@@ -69,7 +69,10 @@ impl<Key : Send + cmp::Eq + hash::Hash> ChainListener for SimpleManyChannelMonit
 	fn block_connected(&self, _header: &BlockHeader, height: u32, txn_matched: &[&Transaction], _indexes_of_txn_matched: &[u32]) {
 		let monitors = self.monitors.lock().unwrap();
 		for monitor in monitors.values() {
-			monitor.block_connected(txn_matched, height, &*self.broadcaster);
+			let outputs = monitor.block_connected(txn_matched, height, &*self.broadcaster);
+			for output in outputs {
+				self.chain_monitor.install_watch_script(&output.script_pubkey);
```

Comment:
I think this needs to be install_watch_outpoint not script, as script was meant to monitor a specific tx by a random script, not meant to monitor outpoints which is all we need here (the commitment txn are already being monitored). Sorry for the confusion here, I had to go remind myself of what this stuff was for, heh.

Reply:
Ah yeah didn't get plainly the distinction! Thanks